python/amici/sbml_import.py

```diff
@@ -1640,15 +1775,56 @@ def _parse_trigger(trigger: sp.Expr) -> sp.Expr:
 
     # or(x,y) = not(and(not(x),not(y))
     if isinstance(trigger, sp.Or):
-        return 1-sp.Mul(*[1-_parse_trigger(arg)
+        return 1-sp.Mul(*[1-_parse_heaviside_trigger(arg)
                         for arg in trigger.args])
 
     if isinstance(trigger, sp.And):
-        return sp.Mul(*[_parse_trigger(arg)
+        return sp.Mul(*[_parse_heaviside_trigger(arg)
                         for arg in trigger.args])
 
-    raise SBMLException('AMICI can not parse piecewise functions '
-                        f'with argument {trigger}.')
+    raise SBMLException(
+        'AMICI can not parse piecewise/event trigger functions with argument '
+        f'{trigger}.'
+    )
+
+
+def _parse_event_trigger(trigger: sp.Expr) -> sp.Expr:
+    """
+    Recursively translates a boolean trigger function into a real valued
+    root function
+
+    :param trigger:
+    :return: real valued root function expression
+    """
+    if trigger.is_Relational:
+        root = trigger.args[0] - trigger.args[1]
+
+        # normalize such that we always implement <,
+        # this ensures that we can correctly evaluate the condition if
+        # simulation starts at H(0). This is achieved by translating
+        # conditionals into Heaviside functions H that is implemented as unit
+        # step with H(0) = 1
+        if isinstance(trigger, sp.core.relational.LessThan) or \
+                isinstance(trigger, sp.core.relational.StrictLessThan):
+            # y < x or y <= x
+            return -root
+        if isinstance(trigger, sp.core.relational.GreaterThan) or \
+                isinstance(trigger, sp.core.relational.StrictGreaterThan):
+            # y >= x or y > x
+            return root
+
+    # or(x,y) = not(and(not(x),not(y))
+    if isinstance(trigger, sp.Or):
+        return sp.Mul(*[_parse_event_trigger(arg) for arg in trigger.args])
+
+    if isinstance(trigger, sp.And):
+        raise SBMLException('AMICI can not parse logical AND expressions in '
```

Comment:
why can't we use the same implementation for `_parse_heaviside_trigger`?

Reply:
That's a bit tricky... For Heavisides, we really have to discern "<" and "<=" or ">" and ">=", which will yield different Heaviside functions. For events, we only have one trigger, which doesn't care about this. Moreover, the Heaviside-parser directly clutches a Heaviside function at the trigger, which we don't want for events. So there are slight little difference. I will see, to which degree we can reuse things, but that's not guaranteed to work...

Reply:
Sorry, yeah missed the point here, but I think both `and` and `or` should be implementable using `sp.min` and `sp.max`, see MATLAB interface and comment below.